public/js/main.js

```diff
@@ -288,10 +295,30 @@ function setFileURLs(new_provider) {
       appLoading.stop();
     } else if (lastQuery !== val) {
       animateTop();
-      index.search(val, displayMatchingLibraries);
+      displayItems = 20;
+      queryItems = 100;	
+      lazyScroll = true;
+      index.search(val, {hitsPerPage: queryItems}, function(err, content){
+        cachedQueryResult = content;
+        displayMatchingLibraries(err, content);
+      });
     }
     lastQuery = val;
   }
+  $(window).scroll(_.debounce( function() {
+    if (lazyScroll && $(window).scrollTop() + $(window).height() * 2 >= $(document).height()) {
```

Comment:
Do we need 2 jquery dom calls here (plus the one on the previous line) for `window`? We should be able to simplify that to one.